src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -90,7 +90,7 @@ Copyright (c) .NET Foundation. All rights reserved.
           errors that are only consequences of the root cause identified here.
   -->
   <Target Name="_CheckForUnsupportedTargetFramework"
-          BeforeTargets="_CheckForInvalidConfigurationAndPlatform;RunResolvePackageDependencies;GetFrameworkPaths;GetReferenceAssemblyPaths"
+          BeforeTargets="_CheckForInvalidConfigurationAndPlatform;RunResolvePackageDependencies;GetFrameworkPaths;GetReferenceAssemblyPaths;Restore"
```

Comment:
@zkat @nkolev92 What kind of impact will this have?

As a result of https://github.com/NuGet/NuGet.Client/pull/3548, NuGet restore now fails when you have a (singular) `TargetFramework` property with a semicolon in it, with the message:

> F:\workspace_work\1\s\artifacts\bin\redist\Debug\dotnet\sdk\5.0.100-ci\NuGet.targets(128,5): error : Invalid restore input. Invalid target framework 'unsupported'.

So now we will run our target to check for that case before restore, so we generate a better error message.

Will this cause problems for scenarios where restore could succeed, or partially succeed when there was an invalid TargetFramework?  What about VS restore, which doesn't even run the Restore target?

@clairernovotny, the comment in the updated test seems to indicate that MSBuild.Sdk.Extras might be impacted by this.  Can you comment on that?

Reply:
Sorry, which comment in which test?

Reply:
@clairernovotny https://github.com/dotnet/sdk/pull/12897/files#diff-afae3e8319c39f94ba88319480ed1917

Reply:
NuGet won't restore if it ever detects an "unsupported" framework. 

NuGet could probably try to throw a better error though.

Reply:
As far as hinging it off of restore, it won't run in VS.

Reply:
Fairly sure this was to ensure that we could define the right additional properties from the TFM so that restore would work.

For example, it'd set TPV, TFV and others: https://github.com/novotnyllc/MSBuildSdkExtras/blob/master/Source/MSBuild.Sdk.Extras/Build/Platforms/Windows.targets

Reply:
The above `invalid target framework 'unsupported'` is exactly what we're hitting at the moment in dotnet/aspnetcore and dotnet/aspnetcore-tooling with VS 16.8.0 preview 2.0 (Int Preview) installed. Will the `;Restore` addition here fix that❔

/fyi builds in the dotnet/aspnetcore repo primarily use `dotnet msbuild`. We hit this error there only when attempting to use desktop `msbuild` for `*.vcxproj` and `*.wixproj` projects. Errors go away when only the external preview version of VS (16.8.0 preview 1.0) is visible to the build.

Reply:
(Believe users of the "master" version of VS also see the above error.)

Reply:
Adding `;Restore` will cause a different failure if the issue is that `TargetFramework` is being used instead of `TargetFrameworks` otherwise, this is a different issue.

Reply:
I haven't checked aspnetcore-tooling but aspnetcore does _not_ use `@(TargetFramework)` items with semicolons in them. We do have a mix of projects, some using `@(TargetFramework)` and those that are multi-targeted using `@(TargetFrameworks)`. In addition, the only projects given to desktop `msbuild` in the aspnetcore failure case are `*.proj` and `*.vcxproj` projects. So, if it isn't this bug that's breaking us, what is❔

My error for the aspnetcore case with the Int Preview version of Visual Studio and .NET SDK version `5.0.100-rc.1.20379.10`:
``` text
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(128,5): error : Invalid restore input. Invalid target framework 'unsupported'. Input files: C:\Users\dougbu\.nuget\packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20403.5\tools\Tools.proj. [C:\Users\dougbu\.nuget\packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20403.5\tools\Tools.proj]
```

The Tools.proj project contains `<TargetFramework>net472</TargetFramework>` and doesn't use an SDK. Is not specifying an SDK no longer supported❔

If this fix does address our problem (even if accidentally), will we need more of NuGet/Home#9911 to build successfully or just an update of our .NET SDK to a verioin with this fix in it❔